ui/app/components/operation-logs/index.ts

```diff
@@ -36,6 +40,22 @@ export default class OperationLogs extends Component<OperationLogsArgs> {
 
   addLogLine(t: string, logLine: object) {
     this.logLines = [...this.logLines, { type: t, logLine: logLine }];
+    if (this.isFollowingLogs === false) {
+      this.badgeCount = this.badgeCount + 1;
+    }
+  }
+
+  @action
+  followLogs(element: any) {
+    element.target.parentElement.scroll(0, element.target.parentElement.scrollHeight);
```

Comment:
this is good, but brittle if we change the DOM structure. Could we be using `this.element.querySelector('.output-scroll-y')`?

Reply:
Agreed this should be more specific! I'm having trouble targeting the element with `querySelector` though - I get an error in the console..

`
You attempted to access the 'element' property on a Glimmer Component, but that property does not exist in Ember.js applications, it only exists in Glimmer.js apps.
`

I've just changed that `followLogs` action so that it can be run when the component is initially inserted (so the scroll position is always at the bottom to begin). For some reason it behaves differently when I use `{{did-insert}}`.

I'm a little out of my depth on JS here - would you be able to take a look and see if you can refactor so it's less brittle?

Reply:
ok, so I think the best solution is probably not `querySelector` but instead [`closest('.output-scroll-y')`](https://developer.mozilla.org/en-US/docs/Web/API/Element/closest)

Reply:
@gregone I've refactored using `.closest()` and tidied up the `element.target` conditional a bit. Thanks for the help!